### DISCUSSION TITLE
Is it possible to escape a show rule?

### DISCUSSION THREAD

**Discussion by SpacewaIker** [typst/typst#888] [Q&A] [answered]
I'm wondering if it is possible to "escape" a show rule somehow. What I mean is to use a function one as if no show rule was defined, even though a show rule *is* defined.

For instance, I want all my links in my document to be underlined, so I have: `show link: underline`. However, I want to link something with the chain emoji, so I used `link(<href>, emoji.chain)`. However, now my emoji is underlined, which I do not want. Since underline is a function and not a parameter of `text`, I don't see any way to un-underline something. 

```
// myfile.typ

#show link: underline

#link("link1")
#link("link2")
#link("", emoji.chain)   // <--- uh oh...
#link("link3")
```

Thanks!

**Comment by laurmaedje** [comment, marked as answer]
Not really, unfortunately. You can create a new show rule that starts from scratch, but then you won't get the linking behaviour. Currently, your best bet is probably something like this:
```
#show link: it => {
  if it.body == [#emoji.chain] {
    return it
  }
  underline(it)
}
```

**Reply by SpacewaIker** [reply]
That works! Thanks!
